iaso/tests/api/test_entity_types.py

```diff
@@ -184,3 +185,70 @@ def test_entity_types_are_multitenancy(self):
 
         self.assertEqual(response.status_code, 201)
         self.assertEqual(len(get_response.json()), 1)
+
+    def test_get_mobile_entity_types(self):
+        self.client.force_authenticate(self.yoda)
+
+        # same account as logged user
+        EntityType.objects.create(
+            name="beneficiary", reference_form=self.form_1, account=self.yoda.iaso_profile.account
+        )
+
+        # different account
+        EntityType.objects.create(name="beneficiary", reference_form=self.form_1, account=self.the_gang)
+
+        response = self.client.get("/api/mobile/entitytypes/")
+
+        entity_type_count = EntityType.objects.filter(account=self.yoda.iaso_profile.account).count()
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json()["count"], entity_type_count)
```

Comment:
Testing on "count" is often useless; you have no idea what has been returned.
Also, testing on "count" with only one value is not testing anything. If your code fails the limit and has a limit of 1, you won't see it at all!
Your test doesn't even check if the only entity type returned is the one from your account. What guarantees that the entity type returned is not from the other account?

Also, Since they have the same name and reference_form values, you can't even test the value returned to ensure that.